### PR TITLE
Support LOSSLESS_TRAFFIC_PATTERN and DEFAULT_LOSSLESS_BUFFER_PARAMETER

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it
Support the following tables which were introduced during dynamic buffer calculation
- LOSSLESS_TRAFFIC_PATTERN
- DEFAULT_LOSSLESS_BUFFER_PARAMETER

Signed-off-by: Stephen Sun <stephens@nvidia.com>

#### How I did it

- LOSSLESS_TRAFFIC_PATTERN

|name|type|range|default|mandatory|description|
|---|---|---|---|---|---|
|mtu|uint32|64~10240|1500|true|The maximum packet size of a lossless packet|
|small_packet_percentage|uint32|0~100|100|true|The percentage of small packet|

- DEFAULT_LOSSLESS_BUFFER_PARAMETER

|name|type|range|default|mandatory|description|
|---|---|---|---|---|---|
|default_dynamic_th|int32|-8~7|0|true|The default dynamic_th for all buffer profiles that are dynamically generated for lossless PG|
|over_subscribe_ratio|uint32|-|-|false|The oversubscribe ratio for shared headroom pool.|
||||||Semantically, the upper bound is the number of physical ports but it can not be represented in the yang module. So we keep the upper bound open. As the type is (signed) integer whose lower bound is 0 by nature, we do not need to specify the range.|

#### How to verify it

Run unit test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-acl.yang [hunk 1]
```diff
@@ -0,0 +1,368 @@
+/* this is sonic py yang model */
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-acl.yang]: Please remove this file. It is mistakenly added. ACL is part of template folder

File: src/sonic-yang-models/yang-models/sonic-extension.yang [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module sonic-extension {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-extension.yang]: Please remove this file. It is mistakenly added here and it is part of yang-templates

File: src/sonic-yang-models/yang-models/sonic-policer.yang [hunk 3]
```diff
@@ -0,0 +1,125 @@
+/* this is sonic py yang model */
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-policer.yang]: Please remove this file. It is mistakenly added here and it is part of yang-templates

File: src/sonic-yang-models/yang-models/sonic-types.yang [hunk 4]
```diff
@@ -0,0 +1,259 @@
+module sonic-types {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-types.yang]: Please remove this file. It is mistakenly added here and it is part of yang-templates

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                        length 1..32 {
+                            error-message "Invalid length for the lossless traffic pattern list name.";
+                            error-app-tag lossless-traffic-pattern-invalid-list-name-length;
+                        }
+                    }
+                }
+
+                leaf mtu {
+                    type uint32 {
+                        range 64..10240 {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang]: Is this different from port mtu which has range 1..9216?

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                            error-message "Invaild MTU which should be in [64, 10240].";
+                        }
+                    }
+                    mandatory true;
+                    description "The maximum packet size of a lossless packet";
+                }
+
+                leaf small_packet_percentage {
+                    type uint32 {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang]: Can this be uint8?

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        container LOSSLESS_TRAFFIC_PATTERN {
+
+            description "LOSSLESS_TRAFFIC_PATTERN part of config_db.json";
+
+            list LOSSLESS_TRAFFIC_PATTERN_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang:25]: Do we restrict the name in code with this pattern today? If not I prefer not to introduce any new patterns here as it cause breaks with already configured names which don't align with this pattern or length

File: src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        container DEFAULT_LOSSLESS_BUFFER_PARAMETER {
+
+            description "DEFAULT_LOSSLESS_BUFFER_PARAMETER part of config_db.json";
+
+            list DEFAULT_LOSSLESS_BUFFER_PARAMETER_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang:25]: Do we restrict the name in code with this pattern today? If not I prefer not to introduce any new patterns here as it cause breaks with already configured names which don't align with this pattern or length

File: src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                        }
+                        length 1..32 {
+                            error-message "Invalid length for the default lossless buffer parameter list name.";
+                            error-app-tag default-lossless-buffer-parameter-invalid-list-name-length;
+                        }
+                    }
+                }
+
+                leaf default_dynamic_th {
+                    type int32 {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang]: Can this be int8?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: src/sonic-yang-models/yang-models/sonic-acl.yang (hunk 1 above)
Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-acl.yang]: All 4 files that were added by mistake have been removed.

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang (hunk 7 above)
Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang:25]: No checking in code today but this is to align other QoS tables like [DSCP_TO_TC_MAP](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-dscp-tc-map.yang#L35). Currently, both tables are rendered from the templates provided by the vendor. So it's less likely to have a name breaking this rule

File: src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang (hunk 9 above)
Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang]: what's the benefit of using int8 and what's the common practice in the community?

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang (hunk 5 above)
Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang]: ok. Let's align it with port mtu range

Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang]: fixed

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang (hunk 7 above)
Comment (dgsudharsan) [reply src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang:25]: In this case I prefer not to impose any restrictions that are not present today unless it adds some value or functionality.

File: src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang (hunk 9 above)
Comment (dgsudharsan) [reply src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang]: It is better to align with the smallest possible type that can accomodate the range.

Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-default-lossless-buffer-parameter.yang]: As discussed offline, it's a common practice to use the smallest possible type.
Fixed.

File: src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang (hunk 7 above)
Comment (stephenxs) [reply src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang:25]: As discussed offline, let's keep the pattern checking to follow the current practice in QoS and buffer part.

**Review by dgsudharsan — APPROVED** [review]

**Comment by stephenxs** [comment]
Ci 1594 passed

**Pull request closed without merging**